### PR TITLE
[TODO] Partial review of `Beta` & `MainNet` TODOs during MainNet genesis & migration

### PR DESCRIPTION
**Pull request by Olshansk**
This is a partial (not complete) review of TODOs in the codebase. 

The goal is to merge in parts of it to main and do another followup after.

In particular, this adds and uses the following to help prioritize tasks before, during and after migration:

- `TODO_MAINNET_CRITICAL`
- `TODO_MAINNET_MIGRATION`
- `TODO_POST_MAINNET`

### CODE REVIEW

**Review by red-0ne — APPROVED** [review]
LGTM, thanks for tending to this.

File: x/migration/module/autocli.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func (am AppModule) AutoCLIOptions() *autocliv1.ModuleOptions {
 					Long:           "Claim the account balance of the given Morse account address, by signing the message with the private key of the Morse account.",
 					PositionalArgs: []*autocliv1.PositionalArgDescriptor{{ProtoField: "morse_src_address"}, {ProtoField: "morse_signature"}},
 					Skip:           true, // skipped because autoCLI cannot handle loading & signing using the Morse key.
+					// TODO_MAINNET_MIGRATION(@bryanchriswhite, #1034): Add morse account claiming CLI, incl. examples (see x/supplier/module/autocli.go).
```
Comment (bryanchriswhite) [inline x/migration/module/autocli.go:60]: There is already a TODO for this in x/migration/module/cmd/claim_morse_account.go.
```suggestion
```

File: x/migration/module/cmd/collect_morse_accounts_test.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ import (
 	migrationtypes "github.com/pokt-network/poktroll/x/migration/types"
 )
 
-// TODO_MAINNET(@bryanchriswhite): Add an E2E/integration test using real data.
+// TODO_MAINNET_MIGRATION(@bryanchriswhite): Add an E2E/integration test using real data.
```
Comment (bryanchriswhite) [inline x/migration/module/cmd/collect_morse_accounts_test.go:19]: This is the migration snapshot feature which is now merged and has its own TODOs.
```suggestion
```

File: pkg/client/query/sharedquerier.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func (sq *sharedQuerier) GetParams(ctx context.Context) (*sharedtypes.Params, er
 // GetClaimWindowOpenHeight returns the block height at which the claim window of
 // the session that includes queryHeight opens.
 //
-// TODO_MAINNET(#543): We don't really want to have to query the params for every method call.
+// TODO_MAINNET_MIGRATION(@bryanchriswhite, #543): We don't really want to have to query the params for every method call.
```
Comment (bryanchriswhite) [inline pkg/client/query/sharedquerier.go:92]: Shouldn't these be @red-0ne, given that he's working on historical params (this would be consisten with the diff in x/proof/types/shared_query_client.go, below)?

File: x/tokenomics/keeper/token_logic_modules.go [hunk 4]
```diff
@@ -192,8 +192,6 @@ func (k Keeper) ProcessTokenLogicModules(
 	logger = logger.With("actual_settlement_upokt", actualSettlementCoin)
 	logger.Info(fmt.Sprintf("About to start processing TLMs for (%d) compute units, equal to (%s) claimed", numClaimComputeUnits, actualSettlementCoin))
 
-	// TODO_MAINNET(@red-0ne): Add tests to ensure that a zero settlement coin
-	// due to integer division rounding is handled correctly.
```
Comment (bryanchriswhite) [inline x/tokenomics/keeper/token_logic_modules.go:196]: I'm assuming/confirming that this is remove because it's either already done or not a concern for some other reason?

**Review by bryanchriswhite — CHANGES_REQUESTED** [review]

File: x/migration/module/cmd/collect_morse_accounts_test.go [hunk 5]
```diff
@@ -16,10 +16,6 @@ import (
 	migrationtypes "github.com/pokt-network/poktroll/x/migration/types"
 )
 
-// Note: This test should not be included in CI due to its size (90GB).
-// Users should manually run wget to download the data and verify it on their computer.
-// Reference: https://github.com/pokt-network/poktroll/pull/1039#discussion_r1947036729
-
```
Comment (bryanchriswhite) [inline x/migration/module/cmd/collect_morse_accounts_test.go:23]: Why delete this?

**Review by bryanchriswhite — APPROVED** [review]
